src/main/java/org/nixos/idea/psi/impl/AbstractNixString.kt

```diff
@@ -0,0 +1,51 @@
+package org.nixos.idea.psi.impl
+
+import com.intellij.lang.ASTNode
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.psi.PsiLanguageInjectionHost
+import com.intellij.psi.impl.source.tree.LeafPsiElement
+import org.nixos.idea.psi.NixIndString
+import org.nixos.idea.psi.NixString
+import org.nixos.idea.psi.NixStringLiteralEscaper
+
+
+abstract class AbstractNixString(private val astNode: ASTNode) : PsiLanguageInjectionHost,
+    AbstractNixPsiElement(astNode), NixString {
+
+    override fun isValidHost() = true
+
+    override fun updateText(s: String): NixString {
+        // TODO issue #81 also support single-line strings
+        if (this !is NixIndString) {
+            LOG.info("not a nix ind string")
+            return this
+        }
+        val originalNode = astNode.firstChildNode.treeNext.firstChildNode as? LeafPsiElement
```

Comment:
issue: This method doesn't seem to escape the input, and may therefore also remove existing escape sequences. For some reason, some lines were also duplicated for me after trying it with the following code:

```nix
pkgs.writeShellScript "my-script.sh" ''
  first_of_array=''${ARRAY[0]}
  from_nix=${lib.escapeShellArg someVar}
  ''
```

Reply:
this is a good catch. I was aware this method did not escape the input and would remove existing escape sequences. This is because at the sage where we get the bash text, I have no way to know if the coming `${` is a bash interpolation or a Nix one, so I don't know whether it should be escaped. Likewise for newlines: If I get a `\n` char, should it get converted to an escaped `''\n` or into an actual `\n`?

I think this is where the method of not injecting in the text fragment may have reached its limits :/